src/test/ui/parser/bounds-type.rs

```diff
@@ -8,6 +8,11 @@ struct S<
     T: ?for<'a> Trait, // OK
     T: Tr +, // OK
     T: ?'a, //~ ERROR `?` may only modify trait bounds, not lifetime bounds
+
+    T: ?const Tr, // OK
```

Comment:
I'd like to see a new check-pass test in the RFC test folder for the new syntax under `// check-pass`.
Please also test something like `for<'a> ?const ?Tr + 'a + Trait`.